### ISSUE TITLE
TOR hidden service

### ISSUE THREAD

**Issue by joeuelk** [keybase/keybase-issues#482]
Would be nice to have keybase.io available as a TOR hidden service at some point.

Duck Duck Go does this: http://search.slashdot.org/story/10/09/25/0242244/DuckDuckGo-Search-Engine-Erects-Tor-Hidden-Service

**Comment by markcoker** [comment]
error: Host 'fncuwbiisyh6ak3i.onion:null' refused connection; maybe the server is down

**Comment by maxtaco** [comment]
Seems to work for me.  Are you sure you're running the tor service locally?  we require that.

**Comment by berrythesoftwarecodeprogrammar** [comment]
works completely fine for me

**Comment by jonathancross** [comment]
I am happy the Tor hidden service is up!
Most users will be accessing this hidden service via the [Tor Browser](https://www.torproject.org/projects/torbrowser.html.en).  The Tor Browser comes with NoScript extension turned on by default to prevent leaking private information via JavaScript.  Unfortunately most of the website seems to break without JavaScript.

I would like to help adapt the website to work without JS (eg: [Progressive Enhancement](https://en.wikipedia.org/wiki/Progressive_enhancement) design strategy for example), do you plan to Open Source the website code?

Note: Progressive Enhancement has added benefits in terms of accessibility and could even allow alternative clients like lynx, etc. (even if some functionality must be sacrificed, it would be great to have the basics working).

**Comment by chungy** [comment]
Might I make a suggestion that the Tor hidden service have a TLS certificate? While Tor already nullifies the purposes of encryption, it might be difficult to verify the identity of the hidden service and that you are, in fact, talking to the real Keybase and not a phishing version. It wouldn't be infeasible for an attacker to generate a vanity address where the first 5-6 characters match the real hidden service, and it could be difficult to notice such a swap on the URL bar.

Facebook, as an example, already does this for their hidden service, for the same reason: https://www.facebookcorewwwi.onion/

**Comment by kode54** [comment]
@jonathancross Maybe you may or may not realize this, but the reason for all the Javascript is so that the encryption and decryption of messages, and the decryption of your private keys, all happens on the client side. If you want a script-less client side, then the server has to have access to both your passkey and your private key in the clear, at least while it is using them.

**Comment by jonathancross** [comment]
@kode54 Yes, I'm aware that some functions such as encryption in the browser cannot be done in a secure manner without JavaScript. But the vast majority of the site is read-only info and all of that would work flawlessly without JavaScript (if built correctly). I've built and maintained hundreds of websites for Google that work this way.

Personally, I don't even use the encryption and verification capabilities in the browser as these things work much better from desktop software (and are more secure there). Instead I'm looking up people's keys, usernames, photos, checking social media handles, seeing proof posted, checking who follows who, etc.

**Comment by Mikaela** [comment]
Is the Tor service down currently?

```
└┌(%:~)┌- keybase id mikaela

▶ INFO Error checking feature "ftl": API network error: Get http://keybase5wmilwokqirssclfnsqrjdsi7jdir5wy7y7iu3tanwmtp6oid.onion/_/api/1.0/user/features.json?features=ftl: Can't complete SOCKS5 connection. [tags:chat-trace=xaiKwwOi1fKU]                                                                              
▶ ERROR API network error: doRetry failed, attempts: 3, timeout 5.324s, last err: Get http://keybase5wmilwokqirssclfnsqrjdsi7jdir5wy7y7iu3tanwmtp6oid.onion/_/api/1.0/merkle/path.json?c=1&last=5545093&load_deleted=1&load_reset_chain=1&poll=10&sig_hints_low=11&uid=f6592c0e050967553c04b4e5f475da00: Can't complete SOCKS5 connection. (code 1601)
```

Tor: `Jun 11 13:55:50 sedric Tor-client[5931]: Closed X streams for service [scrubbed].onion for reason resolve failed. Fetch status: No more HSDir available to query.`

**Comment by heronhaye** [comment]
Seems like it, could you try again now?

**Comment by Chewie9999** [comment]
The tor service doesn't seem to be working at the moment.
